cupy/core/_routines_statistics.pyx

```diff
@@ -338,6 +348,52 @@ cdef _mean = create_reduction_func(
      'out0 = a / _type_reduce(_in_ind.size() / _out_ind.size())', None))
 
 
+_count_non_nan = create_reduction_func(
+    'cupy_count_non_nan',
+    ('e->e', 'f->f', 'd->d'),
+    ('(in0 == in0) ? 1.0 : 0.0', 'a + b', 'out0 = a', None), 0)
+
+
+cdef ndarray _nanstd(
+        ndarray a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+    ret = _nanvar(
+        a, axis=axis, dtype=dtype, out=None, ddof=ddof, keepdims=keepdims)
+    return _math._sqrt(ret, dtype=dtype, out=out)
+
+
+cdef ndarray _nanvar(
+        ndarray a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+
+    assert a.dtype.kind != 'c', 'Variance for complex numbers is not ' \
+                                'implemented. Current implemention does not ' \
+                                'convert the dtype'
+
+    _count = _count_non_nan(a, axis=axis, dtype=dtype, keepdims=True)
+    arrmean = a._nansum(axis=axis, dtype=dtype, out=None, keepdims=True)
+    arrmean = arrmean / _count
```

Comment:
To reduce memory consumption, how about writing as follows?
```suggestion
    arrmean /= _count
```

Reply:
I moved this inside `ReductionKernel` to reduce one kernel launch and used `pow` to avoid calculating `sum / _count` twice. Hope it's okay.